lib/shopify_dashboard_plus.rb

```diff
@@ -280,8 +352,12 @@ def get_detailed_revenue_metrics(start_date, end_date = DateTime.now)
                 :total_revenue => total_revenue,
                 :average_revenue => avg_revenue,
                 :daily_revenue => daily_revenue,
+                :max_daily_revenue => max_daily_revenue,
+                :duration => duration,
                 :revenue_per_product => revenue_per_product,
-                :revenue_per_price_point => (revenue_per_price_point.sort_by{|x,y| x.to_f }.to_h rescue {})
+                :revenue_per_price_point => (revenue_per_price_point.sort_by{|x,y| x.to_f }.to_h rescue {}),
+                :discounts_savings => discounts_value,
+                :discounts_quantity => discounts_quantity
               }
 
     return metrics
```

Comment:
It might be worth breaking out all the different metric calculations into a `Report` class of some kind that would have each of them as a method.

Something like

```ruby
class SalesReport
  def initialize(orders)
    @orders = orders
  end

  def sales_per_country
    # ...
  end

  def number_of_sales
    # ...
  end

  def to_h
    {
      sales_per_country: sales_per_country,
      number_of_sales: number_of_sales
    }
  end
end
```

so you can do

```ruby
SalesReport.new(orders).to_h
```